api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,486 @@
+          format: int32
+          required: false
+        - name: batch_flush_timeout
+          in: query
+          description: |
+            Maximum time in seconds to wait for the flushing of each chunk; if the `batch_limit` is reached before this
+            time is reached the messages are immediately flushed to the client.
+            A value of 0 will flush every batch events immediatly as soon as they come in.
+          type: integer
+          format: int32
+          required: false
+        - name: stream_timeout
+          in: query
+          description: |
+            Maximum time in seconds a stream will live before being interrupted. If unspecified will stream
+            indefinately.
+            If this timeout is reached any pending messages (in the sense of `stream_limit`) will be flushed to the
+            client.
+            Stream initialization will fail if `stream_timeout` is lower than `batch_flush_timeout`.
+          type: integer
+          format: int32
+          required: false
+        - name: batch_keep_alive_limit
+          in: query
+          description: |
+            Maximum number of keep-alive messages to get in a row
+            before closing the connection.
+            Unlimited by default.
+          type: integer
+          format: int32
+          required: false
+        - name: Accept-Encoding
+          in: header
+          required: false
+          description: '"If set the client requests the body to be gzip compressed. For Example "Content-Encoding: gzip"'
+          type: string
+      responses:
+        '200':
+          description: |
+            Starts streaming to the client.  
+            Stream format is a continuous series of
+            `SimpleStreamEvent`s separated by `\n`
+          schema:
+            $ref: '#/definitions/SimpleStreamEvent'
+        '400':
+          description: Bad syntax
+          schema:
+            $ref: '#/definitions/Problem'
+        '401':
+          description: Not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: Not found. Typically in case of requesting from a non existing topic of partition.
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Not found. Typically in case of requesting from a non existing topic of partition.
+          schema:
+            $ref: '#/definitions/Problem'
+
+
+# ################################### #
+#                                     #
+#             Definitions             #
+#                                     #
+# ################################### #
+
+definitions:
+  Event:
+    type: object
+    description: |
+      This is the most general representation of an event, that can be processed
+      by Nakadi.
+
+      It should be used as a base definition for all events, that flow through
+      Nakadi by extending attributes of this object type.
```

Comment:
Stronger statements needed here: Defines the generic event structure valid for all events produced / and consumed by Nakadi client applications. 
Must also specify how producers can add event type specific attributes, how they define its scheme and provide this info as contract to any consumers.